### PR TITLE
Resnet gradients cpp

### PR DESCRIPTION
**Pull request by amturati**
@saxenasaurabh 
@alextp 

Remainder of Grads needed for Resnet

### CODE REVIEW

File: tensorflow/c/eager/gradients.cc [hunk 1]
```diff
@@ -263,7 +263,18 @@ Status AddInputList(AbstractOperation* op_,
 Status SetAttrString(AbstractOperation* op_, const char* attr_name,
                      const char* data, size_t length,
                      ForwardOperation* forward_op_) {
-  forward_op_->attrs.Set(attr_name, StringPiece(data, length));
+  std::cout << "inside SetAttrString:" << std::endl;
+  std::cout << "\tdata = " << data << ", len = " <<length << std::endl; 
+  std::cout << "\tattr_name = " << attr_name << std::endl;                     
+  forward_op_->attrs.Set(attr_name, StringPiece(data, length)); // <--this is not setting correctly
```
Comment (amturati) [inline tensorflow/c/eager/gradients.cc:269]: @saxenasaurabh I'm logging the attributes right as they come in and right after we have set the attributes. 
The `attrs.Set()` for Strings and lists aren't working properly; for example here is the output log for the SetAttrString:

```
inside SetAttrString:
        data = SAME, len = 4
        attr_name = padding
        attrs.Get(padding) = 
exiting SetAttrString

Comment (amturati) [reply tensorflow/c/eager/gradients.cc:269]: attrs.Get(padding) is returning empty string for some reason.

File: tensorflow/c/eager/gradients.cc [hunk 2]
```diff
@@ -263,7 +263,18 @@ Status AddInputList(AbstractOperation* op_,
...
+  
+  std::string p;
+  forward_op_->attrs.Get(attr_name, &p);
+  
+  std::cout << "\tattrs.Get(" << attr_name <<") = " << p << std::endl; // <-- p is the empty string
+  std::cout << "exiting SetAttrString" << std::endl;
+
+  // return op_->SetAttrString(attr_name, data, length);
   return op_->SetAttrString(attr_name, data, length);
```
Comment (amturati) [inline tensorflow/c/eager/gradients.cc:278]: However, here the attribute is getting set just fine, strangely.

File: tensorflow/c/eager/resnet_gradients_test.cc [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+void printArr(auto data[], int n) {
+ std::cout << std::endl << "[";
+ for (int i = 0; i < n - 1; i++) {
+   std::cout << data[i] << ", ";
+ }
+ std::cout << data[n - 1] << "]" << std::endl << std::endl;
+}
+
+
+void printTensor(AbstractTensorHandle* t, int size) {
```
Comment (amturati) [inline tensorflow/c/eager/resnet_gradients_test.cc:171]: Will delete these for final merge, just leaving here for debugging purposes

File: tensorflow/c/eager/gradients.cc (hunk 1 above)
Comment (saxenasaurabh) [reply tensorflow/c/eager/gradients.cc:269]: I verified that this is a bug. I have a fix out for review.

Comment (saxenasaurabh) [reply tensorflow/c/eager/gradients.cc:269]: Could you check if a416b4ef0047a0a892cf429c1f41b391e3d5240c fixes this?

Comment (amturati) [reply tensorflow/c/eager/gradients.cc:269]: That fixed it, thanks!

**Comment by vnvo2409** [comment]
@saxenasaurabh 
I will take over this pull request

**Comment by saxenasaurabh** [comment]
> @saxenasaurabh
> I will take over this pull request

Great, thanks!

**Pull request closed without merging**